src/main/java/com/aws/greengrass/smbridge/MessageBridge.java

```diff
@@ -38,44 +50,65 @@
      * @param topicMapping topics mapping
      */
     public MessageBridge(TopicMapping topicMapping) {
-        this.mqttClient = null;
         this.topicMapping = topicMapping;
         this.topicMapping.listenToUpdates(this::processMapping);
         processMapping();
     }
 
-    private void handleMessage(Message message) {
+    private void handleMessage(MQTTMessage message) {
         String sourceTopic = message.getTopic();
         LOGGER.atDebug().kv("sourceTopic", sourceTopic).log("Message received");
 
         LOGGER.atDebug().kv("destinations", sourceDestinationMap).log("Message will be forwarded to destinations");
 
         if (sourceDestinationMap != null) {
-            List<TopicMapping.MappingEntry> destinations = sourceDestinationMap.get(sourceTopic);
-            if (destinations == null) {
-                return;
-            }
-            LOGGER.atDebug().log("HANDLING MESSAGE, NO IMPLEMENTED SM CLIENT");
-            /*
-            for (TopicMapping.MappingEntry destination : destinations) {
-                if (smClient != null) { // TODO: Add SM Client in Message Bridge
-                    Message msg = new Message(sourceTopic, message.getPayload());
-                    try {
-                        smClient.publish(msg);
-                        LOGGER.atDebug().kv("topic", sourceTopic).kv("stream", destination.getStream())
-                                .log("Published message");
-                    } catch (MessageClientException e) {
-                        LOGGER.atError().kv("topic", sourceTopic).kv("stream", destination.getStream())
-                                .log("Failed to publish");
-                    }
+            final Consumer<TopicMapping.MappingEntry> processDestination = destination -> {
+                String stream = destination.getStream();
+                StreamMessage streamMessage;
+                byte[] payload;
+                JSONObject jsonObject = new JSONObject();
+
+                if (destination.isAppendTime()){
```

Comment:
Who configures the boolean for append time? Why this is needed?

Reply:
It's a user-configurable value per mapping, aka it's a member of the topicMappingEntry class. User can specify it in the deployment config. It defaults to false for every mapping except the reserved topic mapping, where it defaults to true. 
It controls whether time is prepended to the message.